azure/azure-release-pipeline.yml

```diff
@@ -58,6 +58,8 @@ extends:
           VARIABLES_KVM: identity_service_config_keycloak
           ENCRYPTED_VARIABLES_KVM: identity-service-variables-encrypted
           ENCRYPTED_VARIABLES_KVM_KEY_CLIENT_SECRET: keycloak_client_secret
+          ENCRYPTED_VARIABLES_KVM_KEY_CLIENT_SECRET_CIS2: keycloak_client_secret_cis2
```

Comment:
Put these variables inside ENCRYPTED_VARIABLES_KVM rather than creating new variables. See https://nhsd-confluence.digital.nhs.uk/display/APM/Per-environment+configuration

Reply:
`keycloak_client_secret_cis2` has already been added to the ENCRYPTED_VARIABLES_KVM ([here](https://github.com/NHSDigital/api-management-infrastructure/blob/334261d70d7778a39aa6670e8126f4074aef5296/ansible/roles/apigee-key-value-maps/vars/main/key_value_maps.yml#L294)) and it's values added to the Secrets Manager.

Adding this variable was done to replicate the way `ENCRYPTED_VARIABLES_KVM_KEY_CLIENT_SECRET` is used for NHS Login, where `KeyValueMapOperations.GetSecureVariables` inserts the value via this jinja template - falling back to `client_secret` if unable to grab this value.

Reply:
My issue is that `KeyValueMapOperations.GetSecureVariablesCIS2` fails to grab the string `keycloak_client_secret_cis2` from the `ENCRYPTED_VARIABLES_KVM_KEY_CLIENT_SECRET_CIS2` jinja template I'm declaring here and defaults to `client_secret`, giving the incorrect client secret for the keycloak client. 

This suggests that `ENCRYPTED_VARIABLES_KVM_KEY_CLIENT_SECRET_CIS2` isn't being built during the pipeline but I'm not sure how to debug/fix this.